### DISCUSSION TITLE
Unexpected listen result with invalid port

### DISCUSSION THREAD

**Discussion by CDoughty08** [uNetworking/uWebSockets.js#350] [Q&A] [answered]
When using `.listen` with only an invalid port
```ts
const port = 'bad port';
uws.App()
  .listen(
    port,
    (token) => {
      if (token) {
        console.log(`Listening on port: ${port}`);
      }
      else {
        console.log(`Failed to start server`);
      }
    });
```
The expected result happens `Failed to start server`

However if a host is specified with an invalid port
```ts
const port = 'bad port';
uws.App()
  .listen(
    '127.0.0.2',
    port,
    (token) => {
      if (token) {
        console.log(`Listening on port: ${port}`);
      }
      else {
        console.log(`Failed to start server`);
      }
    });
```
The result is `Listening on port: bad port` and the server is started as though the port `0` was given.

**Comment by hst-m** [comment, marked as answer]
have you tried using a valid port number instead of a random string? the docs say to pass a number and you are passing a random string https://unetworking.github.io/uWebSockets.js/generated/interfaces/templatedapp.html#listen

**Comment by CDoughty08** [comment]
@hst-m Absolutely, that works as expected. Although IMO if given an invalid port should satisfy the `Listens to hostname & port. Callback hands either false or a listen socket.` rather than falling back as if 0 was given.

Upon a slightly closer look it appears that this binding also passes through undocumented options to the lower level uSockets library as well.

**Comment by ghost** [comment]
This could be improved with better type checks but it's also possible to follow documented APIs.

**Comment by ghost** [comment]
#343 Has the same problematics - types are not checked very strictly
